File: back_igm/core/models/product.py
import json
from typing import Dict,Any

class Product:
    def __init__(self,id:str,title:str,price:float,description:str,image_url:str):
        self.id = id
        self.title = title
        self.price = price
        self.description = description
        self.image_url = image_url

    @staticmethod
    def fromJson(json: Dict[str:Any]) -> "Product":
        return Product(
            id=str(json["id"]),
            title=str(json["title"]),
            price=float(json["price"]),
            description=str(json["description"]),
            image_url=str(json["image_url"])
        )

    def toJson(self) -> json:
        return {
            "id": self.id,
            "title": self.title,
            "price": self.price,
            "description": self.description,
            "image_url": self.image_url
        }

File: front_igm/CustomerView/models/Product.js


export class Product{
    constructor(id,title,price,description,image_url){
        this.id = id; // str
        this.title = title; // str
        this.price = price; // int
        this.description = description; // str
        this.image_url = image_url; // str
    }

    // to save state in db
    save(){}
    // to look for item by id
    static bringByTitle(title) {
        let product = Product();
        return product;
    }
    // to look for all items
    static bringAll(){
        let products = []; // List<Product>
        return products;
    }
    
    // json handlers
    static fromJson(json){
        if (!json) return null;

        return new Product(
            json.id,
            json.title,
            json.price,
            json.description,
            json.image_url
        );
    };
    toJson(){
        return {
            "id": this.id,
            "title": this.title,
            "price": this.price,
            "description": this.description,
            "image_url": this.image_url
        };
    };
    
}